### PR TITLE
Split --include and --exclude into file and archive variants

### PR DESCRIPTION
**Pull request by tsa96**
Split `--include` and `--exclude` into `--include`, `--exclude`, `--include-archives` and `--exclude-archives`, Also splits out some utility functions as the duplicate code was becoming very unwieldy.

### CODE REVIEW

File: src/create.cpp [hunk 1]
```diff
@@ -20,9 +20,13 @@
 
 #include "log.hpp"
 
-static auto enterVPK( std::ofstream& writer, std::string_view vpkPath, std::string_view vpkPathRel, const std::vector<std::regex>& excluded, const std::vector<std::regex>& included ) -> bool;
+static auto enterVPK( std::ofstream& writer, std::string_view vpkPath, std::string_view vpkPathRel, const std::vector<std::regex>& excludes, const std::vector<std::regex>& includes ) -> bool;
+static auto buildRegexCollection( const std::vector<std::string>& regexStrings, const std::string& collectionType ) -> std::vector<std::regex>;
```
Comment (SCell555) [inline src/create.cpp]:
```suggestion
static auto buildRegexCollection( const std::vector<std::string>& regexStrings, std::string_view collectionType ) -> std::vector<std::regex>;
```

File: src/create.cpp [hunk 2]
```diff
@@ -214,3 +194,24 @@ static auto enterVPK( std::ofstream& writer, std::string_view vpkPath, std::stri
 
 	return true;
 }
+
+static auto buildRegexCollection( const std::vector<std::string>& regexStrings, const std::string& collectionType ) -> std::vector<std::regex> {
```
Comment (SCell555) [inline src/create.cpp]:
```suggestion
static auto buildRegexCollection( const std::vector<std::string>& regexStrings, std::string_view collectionType ) -> std::vector<std::regex> {
```

**Comment by tsa96** [comment]
Using this now for Lumper manifest, seems good. @craftablescience will leave to you to merge!

**Review by craftablescience — APPROVED** [review]
lgtm

**Comment by craftablescience** [comment]
thanks!

**Pull request merged**